src/errors.rs

```diff
@@ -1,132 +1,85 @@
-use crate::pok_sig::PoKOfSignatureProofStatus;
-use crate::pok_vc::PoKVCError;
-use failure::{Backtrace, Context, Fail};
+use thiserror::Error;
 #[cfg(feature = "wasm")]
 use wasm_bindgen::prelude::*;
 
-/// Convenience importing module
-pub mod prelude {
-    pub use super::{BBSError, BBSErrorExt, BBSErrorKind};
-}
+use crate::pok_sig::PoKOfSignatureProofStatus;
 
 /// The kinds of errors that can be generated
-#[derive(Debug, Fail, Clone)]
+#[derive(Debug, Error, Clone)]
 pub enum BBSErrorKind {
     /// Error during key generation
-    #[fail(display = "Key Generation Error")]
+    #[error("Key generation error")]
```

Comment:
Is it OK for the capitalization of the error to change or should it remain exactly the same?

Reply:
It shouldn't make a difference AFAICT, since you can check the error kind (as an integer). I also got rid of the separate PoKVCError which is a slightly bigger change.